--- configure.orig	2012-08-29 01:10:48.000000000 +0530
+++ configure	2012-10-07 20:19:31.511160679 +0530
@@ -4663,6 +4663,16 @@ elif [ "$CFG_OPENGL" = "es2" ]; then
         QMakeVar set QMAKE_LIBS_OPENGL_ES2 "`shellArgumentListToQMakeList "$QMAKE_LIBS_OPENGL_ES2"`"
     fi
 
+    if [ -n "$PKG_CONFIG" ] && $PKG_CONFIG --exists sgx-gles2 2>/dev/null; then
+        QMAKE_INCDIR_OPENGL_ES2=`$PKG_CONFIG --cflags-only-I sgx-gles2 2>/dev/null | sed -e 's,^-I,,g' -e 's, -I, ,g'`
+        QMAKE_LIBDIR_OPENGL_ES2=`$PKG_CONFIG --libs-only-L sgx-gles2 2>/dev/null | sed -e 's,^-L,,g' -e 's, -L, ,g'`
+        QMAKE_LIBS_OPENGL_ES2=`$PKG_CONFIG --libs sgx-gles2 2>/dev/null`
+        QMAKE_CFLAGS_OPENGL_ES2=`$PKG_CONFIG --cflags sgx-gles2 2>/dev/null`
+        QMakeVar set QMAKE_INCDIR_OPENGL_ES2 "$QMAKE_INCDIR_OPENGL_ES2"
+        QMakeVar set QMAKE_LIBDIR_OPENGL_ES2 "$QMAKE_LIBDIR_OPENGL_ES2"
+        QMakeVar set QMAKE_LIBS_OPENGL_ES2 "$QMAKE_LIBS_OPENGL_ES2"
+    fi
+
     compileTest unix/opengles2 "OpenGL ES 2.x" $QMAKE_LIBS_OPENGL_ES2 $QMAKE_CFLAGS_OPENGL_ES2
     if [ $? != "0" ]; then
         echo "The OpenGL ES 2.0 functionality test failed!"
@@ -4888,6 +4898,15 @@ if [ "$CFG_EGL" != "no" ] && [ "$CFG_OPE
         QMakeVar set QMAKE_INCDIR_EGL "$QMAKE_INCDIR_EGL"
         QMakeVar set QMAKE_LIBS_EGL "$QMAKE_LIBS_EGL"
     fi       # detect EGL support
+
+    if [ -n "$PKG_CONFIG" ] && $PKG_CONFIG --exists sgx-gles2 2>/dev/null; then
+        QMAKE_INCDIR_EGL=`$PKG_CONFIG --cflags-only-I sgx-gles2 2>/dev/null | sed -e 's,^-I,,g' -e 's, -I, ,g'`
+        QMAKE_LIBS_EGL=`$PKG_CONFIG --libs sgx-gles2 2>/dev/null`
+        QMAKE_CFLAGS_EGL=`$PKG_CONFIG --cflags sgx-gles2 2>/dev/null`
+        QMakeVar set QMAKE_INCDIR_EGL "$QMAKE_INCDIR_EGL"
+        QMakeVar set QMAKE_LIBS_EGL "$QMAKE_LIBS_EGL"
+    fi
+
     if compileTest qpa/egl "EGL" $QMAKE_CFLAGS_EGL $QMAKE_LIBS_EGL; then
         CFG_EGL=yes
     elif [ "$CFG_EGL" = "yes" ]; then
