--- eglfs.pri.orig	2013-06-09 19:06:34.262470038 +0530
+++ eglfs.pri	2013-06-09 22:58:14.104519680 +0530
@@ -1,11 +1,11 @@
 QT += core-private gui-private platformsupport-private
 
-#DEFINES += QEGL_EXTRA_DEBUG
+DEFINES += QEGL_EXTRA_DEBUG
 
 #Avoid X11 header collision
 DEFINES += MESA_EGL_NO_X11_HEADERS
 
 #To test the hooks on x11 (xlib), comment the above define too
 #EGLFS_PLATFORM_HOOKS_SOURCES += qeglfshooks_x11.cpp
 #LIBS += -lX11
 
@@ -20,20 +20,31 @@ SOURCES +=  $$PWD/qeglfsintegration.cpp
 HEADERS +=  $$PWD/qeglfsintegration.h \
             $$PWD/qeglfswindow.h \
             $$PWD/qeglfsbackingstore.h \
             $$PWD/qeglfsscreen.h \
             $$PWD/qeglfscursor.h \
             $$PWD/qeglfshooks.h \
             $$PWD/qeglfscontext.h
 
+#Prabu for v3dfx
+V3DFX_BASE_PATH = /home/prabindh/work1/v3dfx-base
+GRAPHICS_SDK_PATH = /home/prabindh/Graphics_SDK_4_08_00_01/
+HEADERS +=  $$V3DFX_BASE_PATH/api/include/v3dfx_eglimage.h \
+            $$V3DFX_BASE_PATH/api/include/v3dfxbase.h
+INCLUDEPATH += $$V3DFX_BASE_PATH/api/include/ \
+            $$GRAPHICS_SDK_PATH/GFX_Linux_KM/include4
+LIBS += $$V3DFX_BASE_PATH/api/build/v3dfx-base.a
+
+
 QMAKE_LFLAGS += $$QMAKE_LFLAGS_NOUNDEF
 
 INCLUDEPATH += $$PWD
 
+
 !isEmpty(EGLFS_PLATFORM_HOOKS_SOURCES) {
     HEADERS += $$EGLFS_PLATFORM_HOOKS_HEADERS
     SOURCES += $$EGLFS_PLATFORM_HOOKS_SOURCES
     LIBS    += $$EGLFS_PLATFORM_HOOKS_LIBS
     DEFINES += EGLFS_PLATFORM_HOOKS
 }
 
 CONFIG += egl qpa/genericunixfontdatabase
